tests/ServerTest.php

```diff
@@ -1023,6 +1023,162 @@ function ($data) use (&$buffer) {
         $this->assertInstanceOf('InvalidArgumentException', $error);
     }
 
+    public function testInvalidChunkHeaderResultsInErrorResponse()
+    {
+        $error = null;
+        $server = new Server($this->socket);
+        $server->on('request', function ($request, $response) use (&$error){
+            $request->on('error', function ($ex) use (&$error) {
+                $error = $ex;
+            });
```

Comment:
With the curren commits only `expectCallableOnce` will be called and not the assertion of an Exception.